Assets/Scripts/Algorithms/Patrolling/CognitiveCoordinated/CognitiveCoordinatedBase.cs

```diff
@@ -34,25 +40,20 @@ protected override void GetDebugInfo(StringBuilder stringBuilder)
         protected override IComponent[] CreateComponents(IRobotController controller, PatrollingMap patrollingMap)
         {
             _controller = controller;
-            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, Coordinator.GlobalMap);
+            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, _globalMap);
             _collisionRecoveryComponent = new CollisionRecoveryComponent(controller, _goToNextVertexComponent);
-
             return new IComponent[] { _goToNextVertexComponent, _collisionRecoveryComponent };
         }
 
         public override void SetGlobalPatrollingMap(PatrollingMap globalMap)
         {
             base.SetGlobalPatrollingMap(globalMap);
 
-            // Will be set multiple times.
-            // Too bad!
-            Coordinator.GlobalMap = globalMap;
-
-            // We must clear Coordinators knowledge of occupied vertices as we have new robots, and we might just have started a new experiment.
-            // This will break if robots are added mid-experiment.
-            Coordinator.ClearOccupiedVertices();
+            InitializeCoordinator(globalMap);
         }
```

Comment:
This should only be used in the static coordinator version. Remove it from the base.

Reply:
Done